packages/electron/src/electron-main/electron-application.ts

```diff
@@ -0,0 +1,388 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ContributionProvider } from '@theia/core/lib/common/contribution-provider';
+import { MaybePromise } from '@theia/core/lib/common/types';
+import URI from '@theia/core/lib/common/uri';
+import { ElectronSecurityToken } from '@theia/core/lib/electron-common/electron-token';
+import { ChildProcess, fork, ForkOptions } from 'child_process';
+import * as electron from 'electron';
+import { app, BrowserWindow, BrowserWindowConstructorOptions, Event as ElectronEvent, shell, dialog } from 'electron';
+import { realpathSync } from 'fs';
+import { inject, injectable, named } from 'inversify';
+import { AddressInfo } from 'net';
+import * as path from 'path';
+import { Argv } from 'yargs';
+const Storage = require('electron-store');
+const createYargs: (argv?: string[], cwd?: string) => Argv = require('yargs/yargs');
+
+/**
+ * Options passed to the main/default command handler.
+ */
+export interface MainCommandOptions {
+
+    /**
+     * By default, the first positional argument. Should be a file or a folder.
+     */
+    file?: string
```

Comment:
It's not used from anywhere. Do we need it?

Reply:
Yes, it is part of the default CLI args one can pass when starting the application.

Reply:
Do we support it on the `master`? If no, let's remove it; it does not work: https://github.com/eclipse-theia/theia/pull/7964#issuecomment-643187076